plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -2286,7 +2286,7 @@ protected StartAnswer execute(StartCommand cmd) {
                 }
             }
 
-            if (!bootMode.equalsIgnoreCase(ApiConstants.BootType.BIOS.toString())) {
+            if (StringUtils.isNotBlank(bootMode) && !bootMode.equalsIgnoreCase(ApiConstants.BootType.BIOS.toString())) {
```

Comment:
@pavanaravapalli  bootMode compared with boot type. Please check.

Reply:
@sureshanaparti 

here the purpose is to have bootMode initialise with default value and utilise it with uefi enabled vmprofile or normal vmprofile. however the constant which is used confusing so i have done changes and fixed.